bqskit/ir/circuit.py

```diff
@@ -2209,9 +2215,12 @@ def get_unitary_and_grad(
     def instantiate(
         self,
         target: StateLike | UnitaryLike,
-        method: str | None = None,
+        method: str | Instantiater | None = None,
         multistarts: int = 1,
         seed: int | None = None,
+        start_gen: MultiStartGenerator = RandomStartGenerator(),
+        score_fn_gen: CostFunctionGenerator = HilbertSchmidtCostGenerator(),
```

Comment:
```suggestion
        cost_fn_gen: CostFunctionGenerator = HilbertSchmidtCostGenerator(),
```
To be consistent with naming elsewhere.

Reply:
We actually can't name it that, because we use it elsewhere. If we want the ability to specify two cost functions, one for use during instantiation, and one for use to compare instantiation results, we need two names. I had it originally called `cost_fn_gen`, but the instantiaters were complaining that they weren't getting a `cost_fn_gen` since that usually gets captured in the `**kwargs` and passed to the instantiater.

Reply:
Ah got it.